src/course-search/SearchContext.jsx

```diff
@@ -47,7 +45,12 @@ const SearchData = ({ children, searchFacetFilters }) => {
 
   const newQueryString = useMemo(() => {
     const refinementsWithJoinedLists = updateRefinementsFromQueryParams(refinementsFromQueryParams);
-    return qs.stringify(refinementsWithJoinedLists);
+    let refinementString = new URLSearchParams(refinementsWithJoinedLists);
+
+    if (refinementString) {
+      refinementString = refinementString.toString().replace(/[+]/g, '%20');
```

Comment:
Talk to me about what this is doing? Also, I suspect that if we need to do it here, we probably need to do it for *all* query strings, and it should be moved to a function, probably it should be part of `refinementsWithJoinedLists`. My worry is that if we need to do this replacement, we likely need to do others as well, and that's when we should be looking for a library that does this.

Reply:
testing locally this is what I found:
```
**refinements:** 
`{subjects: "Architecture,Computer Science,Humanities"}`

**query-string's stringify:**
`subjects=Architecture%2CComputer%20Science%2CHumanities`

**new URLSearchParams(refinements).toString()**
`subjects=Architecture%2CComputer+Science%2CHumanities`

 **new URLSearchParams(refinements).toString().replace(/[+]/g, '%20')**
 `subjects=Architecture%2CComputer%20Science%2CHumanities`
 
 The headache here was that `new URLSearchPArams().toString() wasn't encoding the `+` for spaces so I needed to manually encode them (since we also don't want to encode the `=`). I used regex because we can potentially have many and `replace('+', '%20')` only grabs the first instance.

Reply:
FWIW I think this is the only time we use `stringify` in frontend-enterprise so no need to replace it anywhere else.

Reply:
and the conditional 
```
if (refinementString) {
```
is there to make sure we're handling cases where this doesn't exist.

Reply:
What happens if you pass URLSearchParams the params with the original arrays instead of the joined lists? It might do that joining for us.

Reply:
Then we might be able to get rid of `refinementsWithJoinedLists` entirely.

Reply:
it seems like specifically the spaces within individual refinements ie `computer science` aren't being joined and those aren't addressed by the original array, sadly